src/basic/process-util.c

```diff
@@ -1727,6 +1728,41 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+        assert_se(sigaddset(&mask, SIGCHLD) >= 0);
+
+        r = posix_spawnattr_init(&attr);
+        if (r < 0)
+                return r;
+        r = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETSIGMASK);
+        if (r < 0)
+                goto out;
+        r = posix_spawnattr_setsigmask(&attr, &mask);
+        if (r < 0)
+                goto out;
+
+        r = posix_spawn(&pid, path, NULL, &attr, argv, envp);
```

Comment:
This (also possibly others in the above) does not return negative errno, but positive errno, and returns zero on success, if the man page is correct.

Reply:
good catch, missed that